src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureUnitTests.cs

```diff
@@ -0,0 +1,1019 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. 
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.DetectPreviewFeatureAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public partial class DetectPreviewFeatureUnitTests
+    {
+        private const string setupPreviewAttribute = @"
+    [AttributeUsage(AttributeTargets.Assembly |
+                AttributeTargets.Module |
+                AttributeTargets.Class |
+                AttributeTargets.Struct |
+                AttributeTargets.Delegate |
+                AttributeTargets.Interface |
+                AttributeTargets.Enum |
+                AttributeTargets.Constructor |
+                AttributeTargets.Method |
+                AttributeTargets.Property |
+                AttributeTargets.Field |
+                AttributeTargets.Event, Inherited = false)]
+    public sealed class RequiresPreviewFeaturesAttribute : Attribute
+    {
+        public RequiresPreviewFeaturesAttribute() { }
+    }
+
+";
+
+        [Fact]
+        public async Task TestPreviewMethodUnaryOperator()
+        {
+            var csInput = @" 
+using System;
+namespace Preview_Feature_Scratch
+{" + setupPreviewAttribute +
+@"
+    public class Program
+    {
+        static void Main(string[] args)
+        {
+            var a = new Fraction();
+            var b = [|+a|];
+        }
+    }
+
+    public readonly struct Fraction
+    {
+        [RequiresPreviewFeatures]
+        public static Fraction operator +(Fraction a) => a;
+    }
+}
```

Comment:
What should happen if `Fraction` is defined in a generated file, but `Program` is not? Should there be a diagnostic?

i.e, should we use `GeneratedCodeAnalysisFlags.Analyze`?

Reply:
Hmm interesting. I hadn't seen that enum before. I think having `Analyze` makes sense. Have you seen a unit test that tests this behavior?

Reply:
@pgovind Do you mean testing generated code? I think you need to pass two sources, and add `// <auto-generated/>` comment on one of them.

Reply:
> Have you seen a unit test that tests this behavior

I don't think we need to test if the generated code analysis working